common/src/main/java/org/wso2/testgrid/common/config/Script.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+
+package org.wso2.testgrid.common.config;
+
+import java.io.Serializable;
+import java.util.Properties;
+
+/**
+ * Defines a model object for a provided custom script.
+ */
+public class Script implements Serializable {
+
+    private static final long serialVersionUID = 6547552538295691010L;
+
+    private String name;
+    private ScriptType type;
+    private Phase phase;
+    private String description;
+    private String file;
+    private Properties inputParameters = new Properties();
+
+    /**
+     * This defines the supported executable script types.
+     */
+    public enum Phase {
+        /**
+         * Defines the create-infra shell script type.
+         */
+        CREATE("Create"),
+
+        /**
+         * Defines the deploy shell script type.
+         */
+        DEPLOY("Deploy"),
+
```

Comment:
Don't we need an undeploy phase as well?

Reply:
Created https://github.com/wso2/testgrid/issues/501 to track.